solr/core/src/java/org/apache/solr/handler/admin/ClusterStatus.java

```diff
@@ -356,28 +346,28 @@ public static Map<String, Object> postProcessCollectionJSON(Map<String, Object>
     return collection;
   }
 
-  private Map<String, Object> collectionPropsResponse(
+  private Map<String, Object> buildResponseForCollection(
       DocCollection clusterStateCollection,
       Map<String, List<String>> collectionVsAliases,
       String routeKey,
       List<String> liveNodes,
       Set<String> requestedShards) {
-    Map<String, Object> collectionProps = new HashMap<>();
     Map<String, Object> collectionStatus;
+    Set<String> shards = new HashSet<>(requestedShards);
     String name = clusterStateCollection.getName();
 
     if (routeKey != null) {
       DocRouter router = clusterStateCollection.getRouter();
       Collection<Slice> slices = router.getSearchSlices(routeKey, null, clusterStateCollection);
       for (Slice slice : slices) {
-        requestedShards.add(slice.getName());
+        shards.add(slice.getName());
       }
```

Comment:
something seems wrong to me here.  If requestedShards is specified, we should use that.  If routeKey is specified, we should use that (to compute the shards).  Or neither (99% of users won't do either).  But we shouldn't do both if for no other reason that it's confusing.

Reply:
We could throw a `SolrException(BAD_REQUEST)` higher up the stack if passed both `shard` and `_route_`. I can update the ref-docs to say you can only do one or the other. But this could break people who are doing both.

I also changed that logic into a single line and `forEach`

Reply:
I recommend that but understand if you leave out of scope.
(I really don't think this is going to break anyone's current usage, BTW)

Reply:
I'll add it in when I come back for the refactor.